src/pip/_internal/utils/entrypoints.py

```diff
@@ -55,9 +55,14 @@ def get_best_invocation_for_this_pip() -> str:
     if exe_are_in_PATH:
         for exe_name in _EXECUTABLE_NAMES:
             found_executable = shutil.which(exe_name)
-            if found_executable and os.path.samefile(
-                found_executable,
-                os.path.join(binary_prefix, exe_name),
+            binary_executable = os.path.join(binary_prefix, exe_name)
+            if (
+                found_executable
+                and os.path.exists(binary_executable)
```

Comment:
Do we want to do the same for `found_executable` just in case, or is it covered by `which()`?

Reply:
I think its covered because if does not exist `which` returns None

Reply:
It seems `which` uses `os.stat` and `F_OK` so this should be covered.

https://github.com/python/cpython/blob/563f0584c9aa80457f3478c2470b603721479eb0/Lib/shutil.py#L1431

Reply:
`shutil.which` won't return a file that doesn't exist -- unless there's a weird race condition due to the file being deleted between when it looked up and when we try to reflect it in the environment.